duffy/legacy/main.py

```diff
@@ -0,0 +1,193 @@
+from typing import Optional
+
+import httpx
+from fastapi import Depends, FastAPI
+from fastapi.responses import JSONResponse
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_401_UNAUTHORIZED,
+    HTTP_403_FORBIDDEN,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from ..version import __version__
+from .api_models import Credentials
+from .operations import obtain_credentials, obtain_destination
+
+description = """
+Duffy is the middle layer running [`ci.centos.org`](https://ci.centos.org). It provisions, tears
+down and rebuilds physical and virtual machines are used to run tests in the CentOS CI Cluster.
+
+This metaclient exposes older endpoints for legacy support and connects them to the path operations
+introduced by the newer version of the Duffy endpoint, until the support for the older endpoints is
+deprecated.
+"""
+
+app = FastAPI(
+    title="Duffy Metaclient for Legacy Support",
+    description=description,
+    version=__version__,
+    contact={"name": "CentOS CI", "email": "ci-sysadmin@centos.org"},
+)
+
+
+@app.get("/Node/get")
+async def get_node(
+    key: str,
+    ver: int = 7,
+    arch: str = "x86_64",
+    count: int = 1,
+    flavor: str = None,
+    cred: Credentials = Depends(obtain_credentials),
```

Comment:
With the changes related to `obtain_credentials` (or `req_credentials`), I don't think you need key here (`APIKeyQuery` takes care of it, see below).

Reply:
We'd need it. This is what happens when it is removed.

![image](https://user-images.githubusercontent.com/49605954/151501814-80a81e92-9112-4e45-9695-05596393340b.png)

It does not expect the `key` parameter anymore and providing it - causes errors.

Reply:
I assume that using `Depends(req_credentials)` (which in turn uses `Security(APIKeyQuery(...))`) takes care of it.

Going by e.g. [this example]()https://medium.com/data-rebels/fastapi-authentication-revisited-enabling-api-key-authentication-122dc5975680), it describes something more complicated but effectively lets people set `access_token` as a header, cookie or query parameter and uses anything in the controller function even though that uses a wrapper dependency under a different name (which is what we would do, too): `api_key: APIKey = Depends(get_api_key)`

Reply:
I've been a bit of a blind person here. `Depends` certainly takes care of the parameter - take a look at the `Lock` icon at the top-right side of my attached screenshot - so I do seem to have added the credentials but not in the field of the said path operation. It is taken care of in the existing commits.